lib/tic_tac_toe_rules.rb

```diff
@@ -0,0 +1,111 @@
+class TicTacToeRules
```

Comment:
This might be a better place to store `all_moves` rather than the Board class.

Reply:
I also don't really get why this is even called "tic tac toe rules". Why is it "rules"? This just seems like a generic board helper class.

Reply:
Jason's comment: I don't believe I need all_moves in this TTTRules class. However, when going back to Board and seeing where I'm using all_moves, I don't really need it in that class either (since I'm doing move validation somewhere else, I don't need a set that contains all valid moves). I modified the code accordingly.

Tony's comment: I called it TicTacToeRules because the class contains all the logic of what a "winning" combination is. I figured these could be described as the "rules" of the game (you move to 3 in a row, then you win). I think another name that might work is "FullSets", but I think it's ok to keep it at TicTacToeRules for now.